### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by HosseinYousefi**

### CODE REVIEW

File: jnigen/test/test_util/test_util.dart [hunk 1]
```diff
@@ -88,9 +88,10 @@ void comparePaths(String path1, String path2) {
   if (diffProc.exitCode != 0) {
     final originalDiff = diffProc.stdout;
     log.warning(
-        "Paths $path1 and $path2 differ, comparing by ignoring space change");
-    final fallbackDiffProc = Process.runSync(
-        "git", [...diffCommand, '--ignore-space-change', path1, path2]);
+        "Paths $path1 and $path2 differ, Running dart format on $path1.");
+    Process.runSync('dart', ['format', path1]);
+    final fallbackDiffProc =
+        Process.runSync("git", [...diffCommand, path1, path2]);
```
Comment (dcharkes) [inline jnigen/test/test_util/test_util.dart:94]: If we run the formatter, shouldn't we just be able to a literal string compare instead?

Unless you want to actually show the diff after formatting, but you're showing the originalDiff two lines below.

**Review by dcharkes — APPROVED** [review]

File: jnigen/test/test_util/test_util.dart (hunk 1 above)
Comment (dcharkes) [reply jnigen/test/test_util/test_util.dart:94]: If we format both files, we don't have to git diff, we can just load the two files as strings and check equality.

Comment (HosseinYousefi) [reply jnigen/test/test_util/test_util.dart:94]: There can be multiple files in a path.

File: jnigen/test/test_util/test_util.dart [hunk 2]
```diff
@@ -88,9 +88,10 @@ void comparePaths(String path1, String path2) {
...
     if (fallbackDiffProc.exitCode != 0) {
       stderr.writeln(originalDiff);
```
Comment (dcharkes) [inline jnigen/test/test_util/test_util.dart:96]: Wouldn't you want to print `fallbackDiffProc.stdout`/`fallbackDiffProc.stderr` here?

Comment (HosseinYousefi) [reply jnigen/test/test_util/test_util.dart:96]: `dart format` is an exceptional thing that happens only when stable != beta. When even after formatting the diff produces a non-zero result, I show whatever happened originally before formatting.

File: jnigen/test/test_util/test_util.dart (hunk 1 above)
Comment (dcharkes) [reply jnigen/test/test_util/test_util.dart:94]: Ah, check.

File: jnigen/test/test_util/test_util.dart (hunk 2 above)
Comment (dcharkes) [reply jnigen/test/test_util/test_util.dart:96]: Shouldn't the fallback diff be more useful? Otherwise you get all the things that are not actually diffs.

Comment (HosseinYousefi) [reply jnigen/test/test_util/test_util.dart:96]: I understand your point. My logic is: if beta fails, stable will surely fail too, so we get the good diff there. `dart format` is only a hack to fix the CI in our current situation. And if this is no longer the case, we're falling back to the original way. (As if we never had the `dart format`).

**Pull request merged**